comms/src/connectivity/test.rs

```diff
@@ -187,28 +188,19 @@ async fn add_many_managed_peers() {
     let connection_states = connectivity.get_all_connection_states().await.unwrap();
     assert_eq!(connection_states.len(), 10);
 
-    for i in 0..5 {
-        let state = connection_states
-            .iter()
-            .find(|s| s.node_id() == &peers[i].node_id)
-            .unwrap();
+    for i in peers.iter().take(5) {
+        let state = connection_states.iter().find(|s| s.node_id() == &i.node_id).unwrap();
         assert_eq!(state.status(), ConnectionStatus::Connected);
         // Check the connection matches the expected peer
-        assert_eq!(state.connection().unwrap().peer_node_id(), &peers[i].node_id);
+        assert_eq!(state.connection().unwrap().peer_node_id(), &i.node_id);
     }
-    for i in 5..6 {
-        let state = connection_states
-            .iter()
-            .find(|s| s.node_id() == &peers[i].node_id)
-            .unwrap();
+    for i in peers.iter().take(6).skip(5) {
```

Comment:
We're not taking 6 but rather 6-5 = 1 - few other places in this file
```suggestion
    for peer in peers.iter().skip(5).take(1) {
```